components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/TiersApiServiceImpl.java

```diff
@@ -106,21 +115,29 @@ public Response tiersTierLevelGet(String tierLevel, Integer limit, Integer offse
      * @return TierDTO matched to the given tier name
      */
     @Override
-    public Response tiersTierLevelTierNameGet(String tierName, String tierLevel, String accept,
+    public Response tiersTierLevelTierNameGet(String tierName, String tierLevel, String xWSO2Tenant, String accept, 
             String ifNoneMatch, String ifModifiedSince) {
-        String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+
+        String requestedTenantDomain = RestApiUtil.getRequestedTenantDomain(xWSO2Tenant);
         try {
+
+            if (!RestApiUtil.isTenantAvailable(requestedTenantDomain)) {
+                throw RestApiUtil.buildBadRequestException("Provided tenant domain '" + xWSO2Tenant + "' is invalid");
+            }
+
             TierDTO.TierLevelEnum tierType;
-            Tier foundTier;
-            APIConsumer apiConsumer = RestApiUtil.getLoggedInUserConsumer();
+            Tier foundTier = null;
             if (!StringUtils.isBlank(tierLevel)) {
                 if (TierDTO.TierLevelEnum.api.toString().equals(tierLevel)) {
```

Comment:
This enum is named lowercase; please see https://github.com/wso2/carbon-apimgt/pull/1357#discussion_r47447320